internal/installer/elasticagent_tar_macos.go

```diff
@@ -80,7 +82,7 @@ func (i *elasticAgentTARDarwinPackage) Exec(ctx context.Context, args []string)
 
 // Enroll will enroll the agent into fleet
 func (i *elasticAgentTARDarwinPackage) Enroll(ctx context.Context, token string) error {
-	cmds := []string{common.GetElasticAgentWorkingPath("elastic-agent"), "install"}
+	cmds := []string{"sudo", common.GetElasticAgentWorkingPath("elastic-agent"), "install"}
```

Comment:
Found! My bad during the conflict resolution 😢 
```suggestion
	cmds := []string{"sudo", common.GetElasticAgentWorkingPath("elastic-agent", "elastic-agent"), "install"}
```